### PR TITLE
decrease memory usage when csv&gzip is on

### LINKED ISSUE #211: [Enhancement] The executor memory usage will be double when write to doris with csv&gz
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhaorongsheng**
# Proposed changes

Issue Number: close #211 211

## Problem Summary:

Use row input stream instead of StringBuilder when write to gzip outputstream.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by gnehil** [comment]
Have you done any tests? If so, can you post the results?

**Comment by zhaorongsheng** [comment]
> Have you done any tests? If so, can you post the results?

@gnehil  We do not add test case for this case. But in our scene the spark executor will be OOM when using original code.

File: spark-doris-connector/src/main/scala/org/apache/doris/spark/load/StreamLoader.scala [hunk 1]
```diff
@@ -375,14 +376,13 @@ class StreamLoader(settings: SparkSettings, isStreaming: Boolean) extends Loader
...
     if (compressType.nonEmpty) {
       if ("gz".equalsIgnoreCase(compressType.get) && format == DataFormat.CSV) {
-        val recordBatchString = new RecordBatchString(RecordBatch.newBuilder(iterator.asJava)
+        val recodeBatchInputStream = new RecordBatchInputStream(RecordBatch.newBuilder(iterator.asJava)
           .format(format)
           .sep(columnSeparator)
           .delim(lineDelimiter)
           .schema(schema)
           .addDoubleQuotes(addDoubleQuotes).build, streamingPassthrough)
-        val content = recordBatchString.getContent
```
Comment (gnehil) [inline spark-doris-connector/src/main/scala/org/apache/doris/spark/load/StreamLoader.scala:384]: I guess that formatting the entire batch of data into a string object takes up extra memory, and copying between streams reduces the memory usage to a read buffer size.

**Comment by gnehil** [comment]
LGTM

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**